src/site/content/en/blog/porting-libusb-to-webusb/index.md

```diff
@@ -0,0 +1,450 @@
+---
+title: 'Porting USB applications to the web. Part 1: libusb'
+subhead: Learn how code that interacts with external devices can be ported to the web with WebAssembly and Fugu APIs.
+description: Learn how code that interacts with external devices can be ported to the web with WebAssembly and Fugu APIs.
+date: 2022-01-30
+hero:
+alt:
+authors:
+  - rreverser
+tags:
+  - blog # blog is a required tag for the article to show up in the blog.
+  - webassembly
+  - capabilities
+  - devices
+---
+
+In [a previous post](/asyncify/), I showed how to port apps using filesystem APIs to the web with [File System Access API](/file-system-access/), WebAssembly and [Asyncify](https://emscripten.org/docs/porting/asyncify.html). Now I want to continue the same topic of integrating [Fugu APIs](https://fugu-tracker.web.app/) with WebAssembly and porting apps to the web without losing important features.
+
+I'll show how apps that communicate with USB devices can be ported to the web by porting [libusb](https://libusb.info/) - a popular USB library written in C - to WebAssembly (via [Emscripten](https://emscripten.org/)), Asyncify and [WebUSB](/usb/).
+
+{% Aside %}
+Fun fact: WebUSB in Chromium also uses libusb under the hood, so what the port achieves is, in fact, one libusb, compiled to WebAssembly, talking to another libusb, shipped as part of the browser, through an intermediate API layer. Isn't the web fun?
+{% endAside %}
+
+## First things first: a demo
+
+The most important thing to do when porting a library, obviously, is choosing the right demo - something that would showcase the capabilities of the ported library, allow you to test it in a variety of ways, and be visually compelling at the same time.
```

Comment:
FWIW it was meant to be tongue-in-cheek joke rather than literal "obviously", so I thought it doesn't fall under that category, but I don't feel strongly about it.

Reply:
I didn't read it as a joke. Tone in notoriously difficult in writing.